packages/react-native/android/src/main/java/backtrace/library/BreadcrumbFileManager.java

```diff
@@ -0,0 +1,119 @@
+package backtraceio.library;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.module.annotations.ReactModule;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.concurrent.locks.ReentrantLock;
+
+@ReactModule(name = BreadcrumbFileManager.NAME)
+public class BreadcrumbFileManager extends ReactContextBaseJavaModule {
+    public static final String NAME = "BreadcrumbFileManager";
+
+    private static final transient String LOG_TAG = BreadcrumbFileManager.class.getSimpleName();
+
+    private ReentrantLock _lock = new ReentrantLock();
+
+
+    private Boolean _enabled = false;
+    private String _sourceFile;
+    private String _fallbackFile;
+    private Integer _maximumBreadcrumbsPerFile;
+
+    private FileWriter _writer;
+
+    private Integer _currentBreadcrumbsLines = 0;
+
+    public BreadcrumbFileManager(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public void use(String sourceFile, String fallbackFile, Integer maximumBreadcrumbs) throws IOException {
+        _sourceFile = sourceFile;
+        _fallbackFile = fallbackFile;
+        _maximumBreadcrumbsPerFile = maximumBreadcrumbs / 2;
```

Comment:
The class is reimplemented in react-native for react native purposes. Not sure if this is something we should do tbh. If we won't nothing will happen.

Reply:
The implementation anyway will be different in react and node. In addition to that we do not support locks and I strongly believe we need to handle it on native to avoid hard to reproduce crashes.

In node we use stream writer. The same class instance is impossible to return from native. Implementing something similar might a bad idea since it will cost us time.

I prefer to stick to the implementation on the native side. The code is easy to understand and it's easy to maintain. I do not want to create a file system library.

Reply:
Summary:

- We divide number of breadcrumbs per file in node. The implementation should be similar
- we stick to the native layer implementation on react-native however with some small changes like returning boolean from use and renaming the file